backend/src/test/java/com/zeros/backend/service/RouteServiceImplTest.java

```diff
@@ -0,0 +1,35 @@
+package com.zeros.backend.service;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.google.maps.model.TravelMode;
+import com.zeros.backend.BackendApplication;
+
+@SpringBootTest(classes = BackendApplication.class)
```

Comment:
This test is flakey . If you don't have network it will fail

Reply:
this test case is made to test the true integration with the Google API and it is necessary

Reply:
What actually you want to test is that the Google API is up , not that is returning good results as expected.